presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimateMath.java

```diff
@@ -14,86 +14,196 @@
+            newSymbolStats.setDistinctValuesCount(newDistinctValuesCount);
+
+            // range
+            double supersetLow = supersetSymbolStats.getLowValue();
+            double supersetHigh = supersetSymbolStats.getHighValue();
+            double subsetLow = subsetSymbolStats.getLowValue();
+            double subsetHigh = subsetSymbolStats.getHighValue();
+            if (isEmptyRange(supersetLow, supersetHigh)) {
+                newSymbolStats.setLowValue(NaN);
+                newSymbolStats.setHighValue(NaN);
+            }
+            else if (isEmptyRange(subsetLow, subsetHigh) || isUnknownRange(subsetLow, subsetHigh)) {
+                newSymbolStats.setLowValue(supersetLow);
+                newSymbolStats.setHighValue(supersetHigh);
+            }
+            else {
+                verify(supersetLow <= subsetLow, "supersetLow [%s] must be less than or equal to subsetLow [%s]", supersetLow, subsetLow);
+                verify(supersetHigh >= subsetHigh, "supersetHigh [%s] must be greater than or equal to subsetHigh [%s]", supersetHigh, subsetHigh);
+
+                double newLow = supersetLow;
+                double newHigh = supersetHigh;
+
+                if (supersetLow == subsetLow && supersetHigh != subsetHigh) {
+                    newLow = subsetHigh;
+                }
+                if (supersetHigh == subsetHigh && supersetLow != subsetLow) {
+                    newHigh = subsetLow;
+                }
+
+                newSymbolStats.setHighValue(newHigh);
+                newSymbolStats.setLowValue(newLow);
+            }
+
+            result.addSymbolStatistics(symbol, newSymbolStats.build());
+        });
+
+        return result.build();
     }
 
-    public static PlanNodeStatsEstimate differenceInStats(PlanNodeStatsEstimate left, PlanNodeStatsEstimate right)
+    private static boolean isEmptyRange(double low, double high)
     {
-        return differenceInStatsWithRangeStrategy(left, right, StatisticRange::subtract);
+        if (isNaN(low) || isNaN(high)) {
+            checkArgument(isNaN(low) && isNaN(high), "low and high are both expected to be NaN. low: %s, high: %s.", low, high);
+            return true;
+        }
+        return false;
     }
 
-    public static PlanNodeStatsEstimate differenceInNonRangeStats(PlanNodeStatsEstimate left, PlanNodeStatsEstimate right)
+    private static boolean isUnknownRange(double low, double high)
     {
-        return differenceInStatsWithRangeStrategy(left, right, ((leftRange, rightRange) -> leftRange));
+        return low == NEGATIVE_INFINITY && high == POSITIVE_INFINITY;
     }
 
-    private static PlanNodeStatsEstimate differenceInStatsWithRangeStrategy(PlanNodeStatsEstimate left, PlanNodeStatsEstimate right, RangeSubtractionStrategy strategy)
+    public static PlanNodeStatsEstimate computeSymmetricDifferenceStats(
```

Comment:
This method is a combo. One cannot deduce what it does from the method name (it is in fact Inclusion–exclusion `OR` implementation). It can be splited into: `cap` https://github.com/prestodb/presto/pull/12060/commits/1d487a5e213128c6c572a2420ccd5a6820fd361f#diff-59fa6c2264eb72e0587e7e77ce155e59R151, 
`addNonOverlapping` https://github.com/prestodb/presto/pull/12060/commits/1d487a5e213128c6c572a2420ccd5a6820fd361f#diff-59fa6c2264eb72e0587e7e77ce155e59R107 and `subtractSubset` (already exists). Each with well defined semantics.

Reply:
Symetric difference is a standard set operation: https://en.wikipedia.org/wiki/Symmetric_difference